rules/event-scopes.js

```diff
@@ -0,0 +1,120 @@
+const {
+  is,
+  isAny
+} = require('bpmnlint-utils');
+
+const {
+  getEventDefinition,
+} = require('./helper');
+
+module.exports = function() {
```

Comment:
So from how I see it we want to generally warn on the following cases:

* two interrupting catching events conflict with each other (i.e. it is not clear at run-time which one would fire, and interrupt)

Let's model it exactly like that.

Example we can expand that rule with: 

* Two interrupting message events, catching the same message
* Two interrupting conditional events, catching the same condition
* Two interrupting signal events, catching the same signal
* ...

(All according to the scope logic, of course).

I suggest that we extract the fairly complicated logic into dedicated utilities, and test it separately (and more efficiently.

I suggest that we allow explicit configuration of _catchAll_ behavior (i.e. do we want it to be an issue or not). It is supported in some engines only.

Reply:
I suggest we look into this together.